BYS: Move a few things around in /launching-software/infrastructure

dylanirlbeck opened the pull request:
Closes #2549 

I realized there wasn't as much to edit as I thought. It is likely that without using this guide in the context of actual engineering work, I can't hope to really think of a better structure. I am instead closing the linked issue by only proposing a few minor changes.

EDIT: With Aaron's help, this PR also modifies the Table of Contents rendering logic. Now, only headers that are less than 4 levels deep will be rendered.

dylanirlbeck committed:
Init

adborden reviewed (changes requested):
Maybe I'm misunderstanding. Are you suggesting we leave the headings as bold? I don't think we want to close the issue without addressing this.

Commit:
Limit TOC to 2 levels of depth

TOC of 4 levels resulted in a very long and unhelpful Table of Contents
navigation. Let's limit it to 2 for now.

Commit:
Use reference links

Avoid duplicating the URLs for #infrastructure

dylanirlbeck committed:
Merge 8a2702960a4e59b491996eb4b523e60569f530da into 5078bcb762c518b40c8e1a4aadc3b0d25eb2506d

Commit:
Format code

dylanirlbeck committed:
Reverting header changes
